File: cgi-bin/AddDeveloper.java
import java.sql.*;
import oracle.jdbc.pool.OracleDataSource;

public class AddDeveloper {
    public static void main(String[] args) throws SQLException {
        String name = args[0];

        String database = System.getenv("ORACLE_HOST");
        String user = System.getenv("ORACLE_USER");
        String password = System.getenv("ORACLE_PASSWORD");

        // Open an OracleDataSource and get a connection
        OracleDataSource ods = new OracleDataSource();
        ods.setURL("jdbc:oracle:thin:@" + database);
        ods.setUser(user);
        ods.setPassword(password);
        Connection conn = ods.getConnection();

        String status = "";
        try {
            // Create and compose statement
            Statement stmt = conn.createStatement();
            String query = String.format("INSERT INTO developers (name) VALUES ('%s')", name);

            // Insert the data into the developers table
            stmt.executeUpdate(query);
            stmt.close();

            status = "success";
        } catch (SQLException e) {
            System.err.println(e);
            status = "error";
        } finally {
            System.out.printf("{\"status\": \"%s\", \"name\": \"%s\"}", status, name);

            // Commit and close the connection
            conn.close();
        }
    }
}

File: cgi-bin/AddDeveloper.py
#!/usr/bin/env python

import cgi
import cx_Oracle as Oracle
import json
import os
from dotenv import load_dotenv, find_dotenv


def main():
    credentials = {
        'user': os.environ.get('ORACLE_USER'),
        'password': os.environ.get('ORACLE_PASSWORD'),
        'dsn': os.environ.get('ORACLE_HOST'),
    }

    form = cgi.FieldStorage()

    # Get the name of the new developer submitted through the form
    name = form.getvalue('dev_name', '')

    status = 'error'

    connection, cursor = None, None
    try:
        with Oracle.connect(**credentials) as connection:
            cursor = connection.cursor()

            sql = """\
                INSERT INTO developers (name)
                VALUES (:name)
            """

            cursor.execute(sql, name=name)

        status = 'success'
    except Oracle.DatabaseError as e:
        print e
    finally:
        if cursor is not None:
            cursor.close()

        print json.dumps({'status': status, 'name': name})
    # end finally
# end def main()


if __name__ == '__main__':
    # Load in environment variables
    load_dotenv(find_dotenv())

    main()
